### ISSUE TITLE
HS Siphon Build issue (development)

### ISSUE THREAD

**Issue by atereshkov** [Tune389/Tap-Titans-2-Progress-Bot#80]
Hey guys, hey @chrisreyn 

So I was able to test the script and I've noticed that the main functionality is working and it's good, thanks a lot @chrisreyn.

But, I found a problem with the skills leveling.
Even if you set up
`var #justUnlockSkills 1`
the script always try to level up the other skills (Midas, Sword, SC..) so I guess there is some problem.

And it can stuck for a ~minute with this toast message like rgb, colors, etc: https://imgur.com/a/RoBr7

**Comment by BblCTPEJl** [comment]
Bot does not close the offer of fairies

**Comment by atereshkov** [comment]
@BblCTPEJl it works as for me

**Comment by BblCTPEJl** [comment]
At me it accepts and advertising begins, what to me to do?

**Comment by BblCTPEJl** [comment]
Thenks

**Comment by Hassliebe** [comment]
Also, ive seen the bot spend diamonds to hit the clanquest boss again, even if it is turned off.

**Comment by chrisreyn** [comment]
feel free to remove the toast message commands. i didnt remove them for debugging purposes but you can do that yourself. it gets "stuck" on them because it needs to toast all the messages haha

as for justUnlockSkills == 1, make sure to switch these 2 (for some reason order matters because whatever is the latter doesnt work)

`elseif #justUnlockSkills == 1`
`if #skillCheckStepper < 6`
`if #unlocked < 4`
`#unlocked = #unlocked + 1`
`touchPress 0 459 755`
`sleep #btnDelay`
`#time = #time + #btnDelay`
`goto :endLvlSkill`
`elseif #unlocked >= 4`
`if #minutes < #maxSkillsAtXMinutes`
`goto :endLvlSkill`
`else`
`if #unlocked < 8`
`touchPress 0 459 755`
`sleep #btnDelay`
`#time = #time + #btnDelay`
`endif`
`goto :maxSkill`
`endif`
`endif`
`endif`
***make sure you only move 2 endif, dont remove/move the 3rd one


`elseif #justUnlockSkills == 0`
`goto :maxSkill`

^^^switch the places of those 2!!!

and i guess remove this part (UNLESS you want to max your skills later on towards the end of your run, this is found within the 1st spoiler-ed code above)

`elseif #unlocked >= 4`
`if #minutes < #maxSkillsAtXMinutes`
`goto :endLvlSkill`
`else`
`if #unlocked < 8`
`touchPress 0 459 755`
`sleep #btnDelay`
`#time = #time + #btnDelay`
`endif`
`goto :maxSkill`
`endif`
`endif`

**Comment by chrisreyn** [comment]
@Hassliebe I never encountered that problem. If I can't reproduce it, it's gonna be hard to find a way to fix it

Before I worked on this script, it said in elite pvp that sometimes Hiro makes its own problems. It's user based and no one else experiences the problem you are experiencing. I've never touched the clan quest code ever since except for removing the time addition and yesterday, the color check for it to exit loop. Nothing else I changed

**Comment by trant42** [comment]
Hi, do you know how can I make it to level all the skills to level 25 from the beginning?

**Comment by atereshkov** [comment]
@chrisreyn thank you, i'll look into it today

**Comment by Vhivi** [comment]
Hello,

I use the update 4 version of the script and I have a problem, when I'm not strong enough to defeat a boss, we are stuck by the game until we are strong enough or that we have improved the heroes. 
The script at times checks to see if we are stuck or not. At this point it detects that I am blocked and so he clicks on the button to start the boss, but that incomprehensible thing, he continues to click in a loop. 
So I'm seeing the boss scrolling with another mob without being able to type. I have a toast at this time that says "FightBoss1".

**Comment by atereshkov** [comment]
@chrisreyn the issue with the toast msg and stucking seems to have gone, but IDK how to solve the problem with justUnlockSkills. Am I doing smth wrong?

I reworked this to:

```
       #time = #time + 100
	getRGB #colorRed #colorGreen #colorBlue 459 755
	if #colorBlue <= -45
		sleep 10
		#time = #time + 10
		goto :endLvlSkill
	elseif #justUnlockSkills == 1
		if #skillCheckStepper < 6
			if #unlocked < 4
				#unlocked = #unlocked + 1
				touchPress 0 459 755
				sleep #btnDelay
				#time = #time + #btnDelay
				goto :endLvlSkill
			elseif #unlocked >= 4
				//if #minutes < #maxSkillsAtXMinutes
				//	goto :endLvlSkill
				//else
				//	if #unlocked < 8
				//		touchPress 0 459 755
				//		sleep #btnDelay
				//		#time = #time + #btnDelay
				//	endif
				//	goto :maxSkill
				//endif
				goto :endLvlSkill // custom
			endif
		endif
	elseif #justUnlockSkills == 0
		goto :maxSkill
	endif
```

**Comment by rockfreak101** [comment]
Agreed, only but I have consistently seen is the skills upgrade. I looked at the script as but last night and haven't figured you where the issue is, it looks ok at first glance.

**Comment by chrisreyn** [comment]
@atereshkov yes, that should just level each skill once and prevent it from maxing later on at `#maxSkillsAtXMinutes` when using `#justUnlockSkills 1`. It's clicking nearby skills aside from what it's supposed to? I thought that was just the problem for `:maxSkills` section hmmm

@Vhivi i mentioned in a different issue, all you need to do is remove the `goto :PressSkills` line in the `:pressSkills` section, the IF condition with the `toast FightBoss1`. tell me if it is able to successfully click the fight boss button without it needing to loop

**Comment by atereshkov** [comment]
@chrisreyn looks like something wrong elsewhere, could you pls look into this problem (because many of my mates is asking me about this problem and I had mentioned that I'll going to ask you to resolve this annoying bug), thanks a lot.

Just my [final script](http://rgho.st/private/8XrHw7hp6/ba85d243d6cafb22477fc349811c62d5) (if needed).

**Comment by Vhivi** [comment]
@chrisreyn It's good, I confirm that the deletion solves the problem of loop.

**Comment by chrisreyn** [comment]
@Vhivi I guess i'll push that change in the next commit. I just wanted it to make sure it was out of the non-boss fight using the loop, but i guess it was doing more harm than good to a lot of users

@atereshkov like what i mentioned in my comment here https://github.com/Tune389/Tap-Titans-2-Progress-Bot/issues/77#issuecomment-380131948 , since you're using `#justUnlockSkills 1` , probably the line `touchPress 0 459 755` in `:lvlSkill` isn't registering. best to find your own successful clicks. play around with touchUp/touchDown/touchPress and sleep in between each. if you got yours to work, comment here, i will try it out and most probably include it in the next change if it works
